instrumentation/spring/starters/zipkin-spring-boot-starter/build.gradle.kts

```diff
@@ -8,7 +8,7 @@ group = "io.opentelemetry.instrumentation"
 val springBootVersion = "2.6.15"
 
 dependencies {
-  api("org.springframework.boot:spring-boot-starter:$springBootVersion")
+  compileOnly("org.springframework.boot:spring-boot-starter:$springBootVersion")
```

Comment:
@jeanbisutti @zeitlinger is this ok? I noticed it was `compileOnly` in the main starter (the reason it's related to this PR is that we're only scanning runtimeClasspath dependencies)

Reply:
`compileOnly` would make more sense for me